src/modules/common/auth0/auth0.service.ts

```diff
@@ -0,0 +1,104 @@
+import { HttpException, Injectable } from '@nestjs/common';
+import fetch from 'node-fetch';
+import * as Sentry from '@sentry/node';
+import Config from '../../../config';
+import type { Auth0Response, EmailVerificationTicket } from './auth0.types';
+
+@Injectable()
+export class Auth0Service {
+  // Get an access token for the Auth0 Admin API
+  async getAdminAccessToken(): Promise<{ access_token: string }> {
+    const options = {
+      method: 'POST',
+      headers: { 'content-type': 'application/json' },
+      body: JSON.stringify({
+        client_id: Config.auth0.backend.CLIENT_ID,
+        client_secret: Config.auth0.backend.CLIENT_SECRET,
+        audience: `https://${Config.auth0.backend.DOMAIN}/api/v2/`,
+        grant_type: 'client_credentials',
+      }),
+    };
+
+    const response = await fetch(
+      `https://${Config.auth0.backend.DOMAIN}/oauth/token`,
+      options,
+    );
+    const json = await response.json();
+
+    return json;
+  }
+
+  // Get the user's profile from auth0
+  async getUserProfile(accessToken: string, userID: string) {
+    const options = {
+      headers: {
+        Authorization: `Bearer ${accessToken}`,
+      },
+    };
+
+    const response = await fetch(
+      `https://${Config.auth0.backend.DOMAIN}/api/v2/users/${userID}`,
+      options,
+    );
+    const json = await response.json();
+
+    return json;
+  }
+
+  // Deletes a user from auth0
+  async deleteUser(accessToken: string, userID: string) {
+    const options = {
+      method: 'DELETE',
+      headers: {
+        Authorization: `Bearer ${accessToken}`,
+      },
+    };
+
+    const response = await fetch(
+      `https://${Config.auth0.backend.DOMAIN}/api/v2/users/${userID}`,
+      options,
+    );
+
+    return response;
+  }
+
+  async createVerificationEmailTicket(
+    accessToken: string,
+    auth0UserId: string,
+  ) {
+    try {
+      const [provider, userId] = auth0UserId.split('|');
+      const payload = {
+        result_url: Config.urls.CLIENT_BASE_URL,
+        user_id: auth0UserId,
+        identity: { user_id: userId, provider },
```

Comment:
Out of curiousity, why are you stripping of the first part of the userId?

It looks like you refer to userId as the part behind the `|`, while at Auth0 we refer to a userId including what's in front of the `|`. 

Looking at https://auth0.com/docs/manage-users/user-accounts/identify-users, it seems to be that we guarantee the userId to be unique in a tenant (which means, including the part before the `|`). I am not sure there is any guarantee the part behind the `|` is unique across multiple providers.

That said, I am not entirely sure here. But if u don't need to strip it, I wouldn't strip it.

Reply:
First, this is what works for me :)
The payload's `user_id` is indeed the full userId.
The `identify` is the combination of stripped userId and the provider.
Isn't that combination guarantee unique userId?
I can try passing the full userId in identify if you think it ahould work.

Reply:
I just wanted to call it out that you are aware. Not asking to change anything. The most important part is that it works for u, just as I mentioned ... If u don't need the stripping, I wouldn't do it. But if u need it, u need it.

Reply:
> Isn't that combination guarantee unique userId?

Yes but the prefix is part of the userId for us. So seeing it stripped of while still referring to it as a userId can be confusing if you are used to Auth0 UserId's.

Reply:
Are you saying I should use the full userid or should I name it differently? If 2, any suggestion?

Reply:
Are you saying I should use the full userid or should I name it differently? If 2, any suggestion?

Reply:
Thank you!